src/query/service/src/table_functions/inspect_parquet/inspect_parquet_table.rs

```diff
@@ -0,0 +1,276 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::any::Any;
+use std::cmp::max;
+use std::sync::Arc;
+
+use common_catalog::plan::DataSourcePlan;
+use common_catalog::plan::PartStatistics;
+use common_catalog::plan::Partitions;
+use common_catalog::plan::PushDownInfo;
+use common_catalog::table::Table;
+use common_catalog::table_args::TableArgs;
+use common_exception::ErrorCode;
+use common_exception::Result;
+use common_expression::types::DataType;
+use common_expression::types::NumberDataType;
+use common_expression::types::StringType;
+use common_expression::types::UInt64Type;
+use common_expression::types::ValueType;
+use common_expression::BlockEntry;
+use common_expression::DataBlock;
+use common_expression::TableDataType;
+use common_expression::TableField;
+use common_expression::TableSchema;
+use common_expression::TableSchemaRefExt;
+use common_expression::Value;
+use common_meta_app::schema::TableIdent;
+use common_meta_app::schema::TableInfo;
+use common_meta_app::schema::TableMeta;
+use common_pipeline_core::processors::processor::ProcessorPtr;
+use common_pipeline_sources::AsyncSource;
+use common_pipeline_sources::AsyncSourcer;
+use common_sql::binder::resolve_stage_location;
+use common_storage::init_stage_operator;
+use common_storage::read_metadata_async;
+use common_storage::StageFilesInfo;
+use common_storages_fuse::table_functions::string_literal;
+
+use crate::pipelines::processors::port::OutputPort;
+use crate::pipelines::Pipeline;
+use crate::sessions::TableContext;
+use crate::table_functions::TableFunction;
+
+const INSPECT_PARQUET: &str = "inspect_parquet";
+
+pub struct InspectParquetTable {
+    uri: String,
+    table_info: TableInfo,
+}
+
+impl InspectParquetTable {
+    pub fn create(
+        database_name: &str,
+        table_func_name: &str,
+        table_id: u64,
+        table_args: TableArgs,
+    ) -> Result<Arc<dyn TableFunction>> {
+        let args = table_args.expect_all_positioned(table_func_name, Some(1))?;
+        let file_path = String::from_utf8(
+            args[0]
+                .clone()
+                .into_string()
+                .map_err(|_| ErrorCode::BadArguments("Expected string argument."))?,
+        )?;
+        if !file_path.starts_with('@') {
+            return Err(ErrorCode::BadArguments(format!(
+                "stage path must start with @, but got {}",
+                file_path
+            )));
+        }
+        let uri = file_path.strip_prefix('@').unwrap().to_string();
```

Comment:
```suggestion
        let uri = file_path;
```

Reply:
the `InspectParquetTable` will be `create` twice, the 2nd time use the TableInfo.

Reply:
```rust
Test finished, fail fast enabled, 1 out of 157 records failed to run
0: query failed: mysql client error: Server error: `ERROR HY000 (1105): UnknownStage. Code: 2501, Text = Unknown stage @data.'
[SQL] select * from inspect_parquet('@data/parquet/tuple.parquet')
at tests/sqllogictests/suites/stage_parquet/inspect_parquet:1
```
@youngsofun   error again , why it be created twice ?  why InferSchemaTable not ?

Reply:
created twice partly because we need a serializable `Info` for distributed exec.
the new error because  `stage name` is `data` not `@data`.

Reply:
@youngsofun so i need add lock , before i parse the file_path?

Reply:
no need for lock.

```
create() -> TableFunction -- .as_table()- > Table ->  get_table_info() -> TableInfo -- create()  -->  TableFunction
```

Reply:
@youngsofun so , how to fix ?